### ISSUE TITLE
3 Audio tracks

### ISSUE THREAD

**Issue by hjone72** [clinton-hall/nzbToMedia#951]
Hi,

I am wanting to have 3 streams of audio per file. DTS, AC3 and AAC.
However only the first two streams are getting created.

My settings are as follows:
```
OutputAudioCodec: dts
AudioCodecAllow: dts

OutputAudioCodec2: ac3
AudioCodecAllow2: ac3

OutputAudioCodecOther: aac
AudioCodecAllowOther: libfaac
```

The output from the above settings is:
`nice -n10 ionice -c2 -n4 /usr/bin/ffmpeg -loglevel warning -fflags +genpts -i /home/downloads/nzb/sickrage/Modern.Family.S06E14.1080p.WEB-DL.DD5.1.h264-NTb-Obfuscated/Modern.Family.S06E14.1080p.WEB-DL.DD5.1.h264-NTb-Obfuscated.mkv -map 0:0 -map 0:1 -map 0:1 -map 0:2 -map 0:3 -c:v copy -c:a:0 dts -strict -2 -c:a:1 copy -c:s copy /home/downloads/nzb/sickrage/Modern.Family.S06E14.1080p.WEB-DL.DD5.1.h264-NTb-Obfuscated/Modern.Family.S06E14.1080p.WEB-DL.DD5.1.h264-NTb-Obfuscated-transcoded.mkv`

Any help is appreciated.

Thanks,

**Comment by clinton-hall** [comment]
so the way I set this up allows you to create 2 audio tracks, using the settings  
```
OutputAudioCodec:
AudioCodecAllow:

OutputAudioCodec2:
AudioCodecAllow2:
```
then I added and extra set of parameters to allow you to keep any other audio tracks (other languages, commentary etc).
so the settings
```
OutputAudioCodecOther
AudioCodecAllowOther
```
are only used when there are other audio tracks present, and will not do any conversion to/replication of the main audio track.

The problem I face here is trying to create an almost infinite number of combinations and parameters without a web service etc that allows the user to simply add more tracks and parameters etc. 

out of curiosity here, what is your need/want to have all 3 audio tracks?

This may be a feature I can add, I just need to figure out the cleanest method.
